pkg/psalabelsyncer/podsecurity_label_sync_controller.go

```diff
@@ -287,19 +291,19 @@ func (c *PodSecurityAdmissionLabelSynchronizationController) syncNamespace(ctx c
 		} {
 			if ns.Labels[typeLabel] != string(psaLevel) || ns.Labels[versionLabel] != currentPSaVersion {
 				changed = true
-				if nsCopy.Labels == nil {
-					nsCopy.Labels = map[string]string{}
+				if nsApplyConfig.Labels == nil {
+					nsApplyConfig.Labels = map[string]string{}
 				}
 
-				nsCopy.Labels[typeLabel] = string(psaLevel)
-				nsCopy.Labels[versionLabel] = currentPSaVersion
+				nsApplyConfig.Labels[typeLabel] = string(psaLevel)
+				nsApplyConfig.Labels[versionLabel] = currentPSaVersion
 
 			}
 		}
 	}
 
 	if changed {
-		_, err := c.namespaceClient.Update(ctx, nsCopy, metav1.UpdateOptions{})
+		_, err := c.namespaceClient.Apply(ctx, nsApplyConfig, metav1.ApplyOptions{FieldManager: controllerName})
```

Comment:
This doesn't do what you think it does.   You should create a new namespace

```
	corev1apply.Namespace("name").WithLabels(map[string]string{
		"one":"val",
		"two":"",
	})
```

That way your field manager only attempts to set what you want.

Reply:
The above sets "two" to an empty string. We don't want that.

From what I can see, though, if we use
```
	corev1apply.Namespace("name").WithLabels(map[string]string{
		"one":"val",
	})
```
all our managed labels but "one" will get removed.

We can use that instead, then, that should actually make the code a bit cleaner, hopefully.

Reply:
:exclamation:  NOTE to self:
We cannot use `controllerName` as the manager because we would not be able to update fields from previous release. We must use `cluster-policy-controller` instead.